src/protocol/codecs/topic.rs

```diff
@@ -93,15 +261,24 @@ impl TopicCodec {
             }
             let key = src.copy_to_bytes(key_len).to_vec();
 
-            // For handled messages so far, the max data size fits into the u8 integer.
-            let val_len = src.get_u8() as usize;
+            // For handled messages so far, the max data size fits into u8/u16 integers.
+            let val_len = if src[0] & 0x80 == 0 {
+                src.get_u8() as usize
+            } else {
+                // The varint functions encode and decode single integer values using a variable-length encoding;
+                // smaller values require fewer bytes. For a specification,
+                // see https://developers.google.com/protocol-buffers/docs/encoding.
+                // Original comment source: https://pkg.go.dev/encoding/binary#pkg-overview
+                let tmp = src.get_u16_le() as usize;
+                ((tmp & 0x7f00) >> 1) | tmp & 0x7f
+            };
```

Comment:
The above part, well, I'm not sure if I could have used some library or something. But I did it manually, and that's the result.